### PR TITLE
Add support for getting public URL for asset

### PR DESCRIPTION
**Pull request by bryan-vh**
## What kind of change does this PR introduce?

Adds support for the `getPublicUrl` method on the StorageFileApi.

## What is the current behavior?

No method to get a public URL for an asset.

## What is the new behavior?

Adds method to get a public URL for an asset.

## Additional context

N/A

### CODE REVIEW

File: Sources/SupabaseStorage/StorageFileApi.swift [hunk 1]
```diff
@@ -184,4 +184,46 @@ public class StorageFileApi: StorageApi {
...
+            query = ""
+        } else {
+            query = "?\(queryStrings.joined(separator: "&"))"
+        }
+        
+        guard let url = URL(string: "\(url)/\(renderPath)/public/\(path)\(query)") else {
+            throw StorageError(message: "badURL")
+        }
+        
+        return url
```
Comment (grdsdev) [inline Sources/SupabaseStorage/StorageFileApi.swift]: I think is better if you use `URLComponents` for building this URL.

**Review by grdsdev — CHANGES_REQUESTED** [review]

File: Sources/SupabaseStorage/StorageFileApi.swift (hunk 1 above)
Comment (bryan-vh) [reply Sources/SupabaseStorage/StorageFileApi.swift]: Sounds good, will update! Thanks for the feedback

File: Sources/SupabaseStorage/StorageFileApi.swift [hunk 2]
```diff
@@ -184,4 +184,38 @@ public class StorageFileApi: StorageApi {
...
+    ///  - download: Whether the asset should be downloaded.
+    ///  - fileName: If specified, the file name for the asset that is downloaded.
+    ///  - options: Transform the asset before retrieving it on the client.
+    public func getPublicUrl(
+        path: String,
+        download: Bool = false,
+        fileName: String = "",
+        options: TransformOptions? = nil
+    ) throws -> URL {
+        guard let url = URL(string: url) else {
```
Comment (grdsdev) [inline Sources/SupabaseStorage/StorageFileApi.swift]:
```suggestion
        guard var components = URLComponents(string: url) else {
```

**Comment by bryan-vh** [comment]
@grsouza updated with your requested changes

File: Tests/SupabaseStorageTests/SupabaseStorageTests.swift [hunk 3]
```diff
@@ -54,4 +54,11 @@ final class SupabaseStorageTests: XCTestCase {
     let data = try await storage.from(id: bucket).download(path: "README.md")
     XCTAssertFalse(data.isEmpty)
   }
+    
+    func testGetPublicUrl() throws {
```
Comment (grdsdev) [inline Tests/SupabaseStorageTests/SupabaseStorageTests.swift:71]: This is failing for me, can you verify?

**Review by grdsdev — CHANGES_REQUESTED** [review]

File: Tests/SupabaseStorageTests/SupabaseStorageTests.swift (hunk 3 above)
Comment (bryan-vh) [reply Tests/SupabaseStorageTests/SupabaseStorageTests.swift:71]: Just fixed the tests and added more cases :)

**Comment by bryan-vh** [comment]
@grsouza any eta on the review?

**Review by grdsdev — APPROVED** [review]

**Comment by grdsdev** [comment]
LGTM!

Thanks @bryan-vh

**Pull request merged**